apollo-federation/src/query_plan/operation.rs

```diff
@@ -1219,6 +1230,20 @@ mod normalized_field_selection {
                 collect_variables_from_directive(dir, variables)
             }
         }
+
+        pub(crate) fn types_can_be_merged(&self, other: &Self) -> Result<bool, FederationError> {
+            let self_definition = self.data().field_position.get(self.schema().schema())?;
+            let other_definition = other.data().field_position.get(self.schema().schema())?;
+            types_can_be_merged(
+                &self_definition.ty,
+                &other_definition.ty,
+                self.schema().schema(),
+            )
+        }
+
+        pub(crate) fn parent_type_position(&self) -> CompositeTypeDefinitionPosition {
+            self.data().field_position.parent()
+        }
```

Comment:
these have to be on the outside of the mod implementation of the `Field` - https://github.com/apollographql/router/blob/1e73b8d2819446cf8eb97805a788ded86ab75f2a/apollo-federation/src/query_plan/operation.rs#L3808